kube-controllers/pkg/controllers/node/hostendpoints.go

```diff
@@ -323,6 +323,12 @@ func (c *autoHostEndpointController) getAutoHostendpointExpectedIPs(node *libapi
 	if node.Spec.Wireguard != nil && node.Spec.Wireguard.InterfaceIPv4Address != "" {
 		expectedIPs = append(expectedIPs, node.Spec.Wireguard.InterfaceIPv4Address)
 	}
+	for _, addr := range node.Spec.Addresses {
```

Comment:
I think we probably need de-duplication here, so that we don't end up with multiple of the same IP in the expectedIPs field. 

I think BGP.IPv4Address especially is likely to be a duplicate, unless that is already handled elsewhere?